src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -0,0 +1,140 @@
+  unmuteLabel?: string;
+  /** Title attribute for the icon when muted, it defaults to `Muted microphone` */
+  mutedIconTitle?: string;
+  /** Title attribute for the icon when unmuted, it defaults to `Microphone` */
+  unmutedIconTitle?: string;
+}
+
+const AudioInputVFControl: React.FC<Props> = ({
+  muteLabel = 'Mute',
+  unmuteLabel = 'Unmute',
+  mutedIconTitle,
+  unmutedIconTitle
+}) => {
+  const meetingManager = useMeetingManager();
+  const [voiceFocusOn, setVoiceFocusOn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [enableVoiceFocus, setEnableVoiceFocus] = useState(false);
+  const dropdownWithVFOptions = useRef<PopOverItemProps[] | null>(null);
+  const { muted, toggleMute } = useToggleLocalMute();
+  const [device, setDevice] = useState<Device | AudioTransformDevice | null>(meetingManager.selectedAudioInputTransformDevice);
+  const { supportsVoiceFocus, addVoiceFocus } = useVoiceFocus();
+  const audioInputConfig: DeviceConfig = {
+    additionalDevices: true,
+  };
+  const { devices, selectedDevice } = useAudioInputs(audioInputConfig);
+
+  const dropdownOptions: PopOverItemProps[] = devices.map((device) => ({
+    children: <span>{device.label}</span>,
+    checked: isOptionActive(selectedDevice, device.deviceId),
+    onClick: (): Promise<void> =>
+      meetingManager.selectAudioInputDevice(device.deviceId),
+  }));
+
+  useEffect(() => {
+    if (device instanceof VoiceFocusTransformDevice) {
+      setEnableVoiceFocus(true);
+    } else {
+      setEnableVoiceFocus(false);
+    }
+  }, [device]);
+
+  useEffect(() => {
+    meetingManager.subscribeToSelectedAudioInputTransformDevice(setDevice);
+
+    return (): void => {
+      meetingManager.unsubscribeFromSelectedAudioInputTransformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (supportsVoiceFocus) {
+      const vfOption: PopOverItemProps = {
+        children: <><PopOverSeparator />{isLoading && <Spinner />} {enableVoiceFocus ? <span>Disable voice focus</span> : <span>Enable voice focus</span>}</>,
+        checked: enableVoiceFocus,
+        disabled: isLoading,
+        onClick: () => {
+          setVoiceFocusOn(current => !current);
+        },
+      };
+      dropdownOptions?.push(vfOption);
+    }
+    dropdownWithVFOptions.current = dropdownOptions;
+  }, [supportsVoiceFocus, enableVoiceFocus, voiceFocusOn, isLoading, dropdownOptions]);
+
+  useEffect(() => {
+    async function onVFCheckboxChange() {
+      let current;
+      setIsLoading(true);
+      if (voiceFocusOn) {
+        if (typeof (device) === 'string') {
+          const currentDevice = audioInputSelectionToDevice(device);
+          current = await addVoiceFocus(currentDevice);
+        }
+      } else {
+        if (device instanceof VoiceFocusTransformDevice) {
+          current = device.getInnerDevice();
+        }
+      }
+      if (current) {
+        setDevice(current)
+        meetingManager.selectAudioInputDevice(current);
```

Comment:
So based on `voiceFocusOn` either the `current` will be a normal audio input device else it would be a `AudioTransformDevice`. If it is a `AudioTransformDevice` then should we be calling `meetingManager.selectedAudioInputTransformDevice` and similarly if normal audio input device then call `meetingManager.selectAudioInputDevice`?

Also, should we here subscribe to the `meetingManager.subscribeToSelectedAudioInputDevice` to get the updated device and then the current device will be set through the `setDevice` like you are doing for audio transform in the first `useEffect` in this file? That way we rely on meeting manager to select and do not update the state here explicitly.